### ISSUE TITLE
Panel favorite star

### ISSUE THREAD

**Issue by mahtiankka** [linuxmint/mint20.1-beta#19]
The panel's favorite "star" displays the file, but if you click on it, nothing happens. Tested by Cinnamon.

![Faforite_not_start](https://user-images.githubusercontent.com/48694735/102675731-2ae41f00-41a3-11eb-8f67-a06eb2d08e5c.png)

**Comment by mtwebster** [comment]
How did you add that? From the file manager's menu?  Does the file still exist in the original location?  Can you paste the contents of ~/.xsession-errors when this happens (if it has anything interesting)?

Does this file open properly from the favorites location in nemo? (sidebar->Favorites)

**Comment by mahtiankka** [comment]
I will send a picture and log

A couple of images where the program indicates that kuva2.png does not exist.

When I copy a new image with the same name to the same folder it works immediately.

Perhaps this has happened is that I have removed the image and but has not disappeared from favorites?

I have tried to repeat the hours the problem. I won't get it again. Although the same image is open in all programs and I delete it using Nemo.

I am sending my session-errors file. Hope that helps.

![pix](https://user-images.githubusercontent.com/48694735/102687115-66a7d480-41f5-11eb-8840-277cecdd8d49.png)
![Trashing_photo](https://user-images.githubusercontent.com/48694735/102687116-67406b00-41f5-11eb-9ae1-23bb85dbd32c.png)

[xsession-errors.txt](https://github.com/linuxmint/mint20.1-beta/files/5719196/xsession-errors.txt)

**Comment by mahtiankka** [comment]
I found it!

If I make a folder a favorite. Folder inside the 'Pin' image file. Next, I will delete the folder. The icon remains visible and cannot be opened or changed.

![Pin_fle_delete_problem](https://user-images.githubusercontent.com/48694735/102687542-707f0700-41f8-11eb-8281-d8d1babc2a9b.png)

**Comment by mtwebster** [comment]
Ah now I see. When you directly delete/trash a file that was a favorite, it gets removed from favorites.  If a file gets deleted (or otherwise goes missing) by some other factor, it remains in favorites.  This is done for cases like remote files or folders (like on some server) that may not always be available.  This is a bit of a special case though - you deleted a folder with a favorite in it, should we assume you knew that and want to delete it?

I'll give this some thought - thanks for helping me reproduce it.

**Comment by clefebvre** [comment]
We show files in favorites:// (in nemo) so they can be managed. So even if a path doesn't exist (whether that's because of a removal we weren't aware of, or an HDD that isn't mounted), we still show the favorite so the user can manage it (for instance so that he can remove the file as a favorite).

Now, with that said.. we do that in nemo because that's where people manage favorites. In places where all you can do is access them... if the path isn't accessible, I see no reason to show them.

In other words in that particular case I think the favorite should be in favorites:// (in nemo), but not in the applet nor in the menu.

**Comment by clefebvre** [comment]
I'm moving this to cinnamon since it's not a release blocker.
